src/emmy/expression/compile.cljc

```diff
@@ -369,6 +373,25 @@
               (gensym-fn 'y)))]
     (rec state)))
 
+(defn- cse
+  "Invokes extract-common-subexpressions on `x`.
+
+   The results of the elimination are presented in a map with the keys
+   `:bindings` and `body`.  If no suitable subexpressions were found,
+   the binding list will be empty and body will be unchanged.
+
+   "
+  [x opts]
+  (extract-common-subexpressions
+   x
+   (fn [body bindings]
+     ;; TODO: consistency demands that we do apply-numeric-ops on the
```

Comment:
yup, and it makes sense to do it here instead of before subexpression elimination so we don't repeat all of the work, yeah?

Reply:
done